app/controllers/placements/schools/placements/build_controller.rb

```diff
@@ -0,0 +1,191 @@
+        session[:add_a_placement][:subject_ids] = subject_ids
+      else
+        assign_subjects_based_on_phase
+        render :add_subject and return
+      end
+    when :add_mentors
+      mentor_ids = mentor_ids_params
+      @placement = build_placement
+      @placement.mentor_ids = mentor_ids if mentor_ids.present?
+
+      if @placement.valid_mentor_ids?
+        session[:add_a_placement][:mentor_ids] = mentor_ids
+      else
+        render :add_mentors and return
+      end
+    else
+      raise ActionController::RoutingError, "Not Found"
+    end
+
+    redirect_to public_send(next_step(params[:id]))
+  end
+
+  private
+
+  STEPS = %i[add_phase add_subject add_mentors check_your_answers].freeze
+
+  def reset_session
+    session.delete(:add_a_placement)
+    setup_session
+  end
+
+  def handle_primary_or_secondary
+    session[:add_a_placement][:placement] = @placement.attributes
+    session[:add_a_placement][:phase] = school.phase
+    redirect_to add_subject_placements_school_placement_build_index_path(school_id: school.id)
+  end
+
+  def handle_other
+    redirect_to add_phase_placements_school_placement_build_index_path(school_id: school.id)
+  end
+
+  def setup_session
+    session[:add_a_placement] = {} if session[:add_a_placement].blank?
+  end
+
+  def setup_phase_navigation
+    @enable_phase_navigation = session.dig(:add_a_placement, "enable_phase_navigation")
+  end
+
+  def build_or_retrieve_placement
+    @placement = build_placement
+    session.dig(:add_a_placement, "subject_ids").present? ? build_subjects : @placement.subjects.build
+  end
+
+  def assign_subjects_based_on_phase
+    @phase = session.dig(:add_a_placement, "phase").presence || (school.primary_or_secondary_only? ? school.phase : "Primary")
+    @subjects = @phase == "Primary" ? Subject.primary : Subject.secondary
+    @selected_subjects = find_subjects
+  end
+
+  def find_mentors
+    session[:add_a_placement]["mentor_ids"].present? ? school.mentors.find(session.dig(:add_a_placement, "mentor_ids").compact_blank) : []
+  end
+
+  def find_subjects
+    subject_ids = session.dig(:add_a_placement, "subject_ids")
+    if subject_ids.instance_of?(String)
+      subject_ids.present? ? [Subject.find(subject_ids)] : []
+    else
+      subject_ids.present? ? Subject.find(subject_ids.compact_blank) : []
+    end
+  end
+
+  def initialize_placement
+    placement = Placements::Schools::Placements::Build::Placement.new(school:, status: :draft)
+    build_mentors(placement)
+    build_subjects(placement)
+    placement
+  end
+
+  def build_mentors(placement = nil)
+    placement ||= @placement
+    session[:add_a_placement]["mentor_ids"].compact_blank.each do |mentor_id|
+      placement.mentors << Placements::Mentor.find(mentor_id)
+    end
+  end
+
+  def build_subjects(placement = nil)
+    placement ||= @placement
+    subject_ids = session.dig(:add_a_placement, "subject_ids")
+    if subject_ids.instance_of?(String)
```

Comment:
`placement.subjects << Subject.find(subject_ids)` can handle both the string and the array

Reply:
```suggestion
   placement.subjects << Subject.find(subject_ids)
```
^^ This should be enough